android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -104,9 +106,10 @@ protected void triggerLayoutInflation() {
             public void onClick(View v) {
                 Intent accountDetailsWithQrActivityIntent =
                         new Intent(AccountDetailActivity.this, AccountDetailsWithQrActivity.class);
+                accountDetailsWithQrActivityIntent.putExtra(Utils.COIN_TYPE, mCoinType);
+                accountDetailsWithQrActivityIntent.putExtra(Utils.KEYRING_ID, mKeyringId);
                 accountDetailsWithQrActivityIntent.putExtra(Utils.ADDRESS, mAddress);
                 accountDetailsWithQrActivityIntent.putExtra(Utils.NAME, mName);
-                accountDetailsWithQrActivityIntent.putExtra(Utils.COIN_TYPE, mCoinType);
                 startActivity(accountDetailsWithQrActivityIntent);
```

Comment:
Is there a typesafe way to make sure intent is always started with exact set of variables? Something like a factory with methods to wrap all of these lines and only that factory is allowed to call startActivity()?

Reply:
Can create a static method like [createIntent](https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java#L79) within the destination activity.

Reply:
Can create a static method like [createIntent](https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java#L79) within the destination activity.

Reply:
Thanks for example